src/WingetCreateCore/Common/PackageParser.cs

```diff
@@ -370,7 +370,12 @@ public static Installer CloneInstaller(Installer installer)
         /// <param name="existingInstaller">Existing installer node.</param>
         /// <param name="newInstaller">New installer node.</param>
         private static void UpdateInstallerMetadata(Installer existingInstaller, Installer newInstaller)
-        {
+        {
+            if (IsPreciseInstallerType(newInstaller.InstallerType.Value))
+            {
+                existingInstaller.InstallerType = newInstaller.InstallerType;

```

Comment:
>existingInstaller.InstallerType = newInstaller.InstallerType; [](http://example.com/codeflow?start=16&length=61)

This has more implication than simply overriding the installer type. It might break installer if it's not done correctly. For example, Installer Switches might not be same for different installer types. I suggest we revisit Installer fields to see if some fields need to be reset, we need to also consider what the impact will be for root level fields.